.testing/tools/parse_fms_clocks.py

```diff
@@ -60,18 +60,34 @@ def main():
         print(json.dumps(config))
 
 
-def parse_mom6_param(param_file):
+def parse_mom6_param(param_file, header=None):
     params = {}
     for line in param_file:
         param_stmt = line.split('!')[0].strip()
-        if param_stmt:
-            key, val = [s.strip() for s in param_stmt.split('=')]
-
-            # TODO: Convert to equivalent Python types
-            if val in ('True', 'False'):
-                params[key] = bool(val)
-            else:
-                params[key] = val
+
+        # Skip blank lines
+        if not param_stmt:
+            continue
+
+        # Subparameters
+        if param_stmt[-1] == '%':
+            key = param_stmt.split('%')[-2]
+
+            subheader = key
+            if header:
+                subheader = header + '%' + subheader
+            value = parse_mom6_param(param_file, header=subheader)
```

Comment:
recursive call

Reply:
The function is intentionally recursive, in order to parse the nested `p1%p2%p3% ... %p3%p2%p1` blocks.  Is there a better solution here?